nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyRecipe.groovy

```diff
@@ -207,20 +192,36 @@ class P2ProxyRecipe
     return tokenMatcherForBinary()
   }
 
-  static Builder matchRequestWithExtensionAndName(final String extension, final String name = '.+', final String path = '.+') {
-    new Builder().matcher(
-        and(
-            new ActionMatcher(GET, HEAD),
-            tokenMatcherForExtensionAndName(extension, name, path)
-        ))
+  static TokenMatcher tokenMatcherForBinary() {
+    new TokenMatcher("/{path:.*}/{name:.*}_{version:.*}")
   }
 
   static TokenMatcher tokenMatcherForExtensionAndName(final String extension, final String name = '.+', final String path = '.+') {
-    new TokenMatcher("{path:${path}}/{name:${name}}.{extension:${extension}}")
+    new TokenMatcher("/{path:${path}}/{name:${name}}.{extension:${extension}}")
   }
 
-  static TokenMatcher tokenMatcherForBinary() {
-    new TokenMatcher("{path:.*}/{name:.*}_{version:.*}")
+  static Matcher buildSimpleMatcher(final String path, final String name, final String extension, final AssetKind assetKind) {
+    buildTokenMatcherForPatternAndAssetKind("/{path:${path}}/{name:${name}}.{extension:${extension}}", assetKind, GET, HEAD)
```

Comment:
looks like redundant slash before {path:${path}}, when I tried to install some software my Eclipse throw error that url to proxy couldn't be resolved. Without this slash all is fine installed. Tried to proxy http://findbugs.cs.umd.edu/eclipse/ . Could you please check this?

Reply:
FYI, fixed in https://github.com/sonatype-nexus-community/nexus-repository-p2/pull/26